src/evse_monitor.cpp

```diff
@@ -254,8 +254,16 @@ void EvseMonitor::evseBoot(const char *firmware)
     _heartbeat = RAPI_RESPONSE_OK == ret;
   });
 
+  // ESP32_WiFi handle OpenEvse state, so let first put it to disabled before unlocking
+  if (config_pause_uses_disabled()) {
+    disable();
+  }
+  else {
+    sleep();
+  }
+
```

Comment:
This is the source of the crash in the divert_sim at least, not quite sure why it is not causing a problem on the ESP, but these calls cause a state change, thus calling updateState and ultimately the _energyMeter.increment_switch_counter. However the evseBoot call is before the _energyMeter.begin:
https://github.com/OpenEVSE/ESP32_WiFi_V4.x/blob/master/src/evse_monitor.cpp#L383-L386

Reply:
Ah that makes sense, thanks.
So what would be the best way to prevent evse going to active state first  it hokups the switch at boot ( I can see when Bootlock is disabled the evse switch to active and immediately being set disabled by wifi module.

Reply:
Can you move the energy monitor begin earlier? Maybe pass the EVSE monitor pointer in the constructor?


That being said won't the main loop put the EVSE into the correct state? Having these lines here will only change the state a few fractions of a second sooner right? 

Reply:
Yes evsemanager put it back to correct state, but at startup it will fast switch to active then disabled within a sec. 
Or perhaps we should set evse firmware to default to disabled ?

Reply:
I've moved the EnergyMeter start a bit sooner, seems to fix it.